stubs/tensorflow/tensorflow/strings.pyi

```diff
@@ -0,0 +1,241 @@
+from collections.abc import Sequence
+from typing import Literal, TypeVar, overload
+
+from tensorflow import RaggedTensor, Tensor
+from tensorflow._aliases import StringTensorCompatible, TensorCompatible, UIntTensorCompatible
+from tensorflow.dtypes import DType
+
+_TensorOrRaggedTensor = TypeVar("_TensorOrRaggedTensor", Tensor, RaggedTensor)
+
+@overload
+def as_string(
+    input: TensorCompatible,
+    precision: int = -1,
+    scientific: bool = False,
+    shortest: bool = False,
+    width: int = -1,
+    fill: str = "",
+    name: str | None = None,
+) -> Tensor: ...
+@overload
+def as_string(
+    input: RaggedTensor,
+    precision: int = -1,
+    scientific: bool = False,
+    shortest: bool = False,
+    width: int = -1,
+    fill: str = "",
+    name: str | None = None,
+) -> RaggedTensor: ...
+def bytes_split(input: TensorCompatible | RaggedTensor, name: str | None = None) -> RaggedTensor: ...
+def format(
+    template: str, inputs: TensorCompatible, placeholder: str = "{}", summarize: int = 3, name: str | None = None
+) -> Tensor: ...
+def join(inputs: Sequence[TensorCompatible | RaggedTensor], separator: str = "", name: str | None = None) -> Tensor: ...
+@overload
+def length(input: TensorCompatible, unit: Literal["BYTE", "UTF8_CHAR"] = "BYTE", name: str | None = None) -> Tensor: ...
+@overload
+def length(input: RaggedTensor, unit: Literal["BYTE", "UTF8_CHAR"] = "BYTE", name: str | None = None) -> RaggedTensor: ...
+@overload
+def lower(input: TensorCompatible, encoding: Literal["utf-8", "", " "] = "", name: str | None = None) -> Tensor: ...
```

Comment:
The docstring says `Allowed values are '' and 'utf-8'.` Why do you also accept `" "`?

Reply:
Must have misread the docs, sorry. Fixed in f33b7275b.